eox_tagging/models.py

```diff
@@ -97,14 +103,17 @@ def __get_object_for_this_type(self, object_type, object_id):
             }
 
         if object_type == COURSE_ENROLLMENT_MODEL_NAME:
-            object_id = {
-                "course_id": CourseKey.from_string(object_id.get("course_id")),
-                "user__username": object_id.get("username"),
-            }
 
-        object_instance = ctype.get_object_for_this_type(**object_id)
+            course_id = object_id.get("course_id")
+            if course_id:
+                object_id["course_id"] = CourseKey.from_string(course_id)
+
+            if "username" in object_id:
+                object_id["user__username"] = object_id.get("username")
```

Comment:
```suggestion
            course_id = object_id.get("course_id")
            username = object_id.get("username")
            object_id = {}
            if course_id:
                object_id["course_id"] = CourseKey.from_string(course_id)

            if username:
                object_id["user__username"] = username
```
There was actually an oversight on my first suggestion and the filter didn't seem to work, this should fix that.